packages/fast-components-styles-msft/src/utilities/density.ts

```diff
@@ -1,6 +1,146 @@
+import { applyTypeRampConfig, TypeRamp, typeRamp } from "../utilities/typography";
 import { DesignSystem, withDesignSystemDefaults } from "../design-system";
+import { CSSRules } from "@microsoft/fast-jss-manager";
 import { toPx } from "@microsoft/fast-jss-utilities";
 
+/**
+ * Returns the type ramp adjustment from the default setting. Collapses
+ * the fact that a single type size is used for multiple densities.
+ *
+ * @param designSystem The design system config
+ */
+export function densityToTypeOffset(designSystem: DesignSystem): number {
+    const offset: number =
+        designSystem.density >= 2 ? 1 : designSystem.density <= -2 ? -1 : 0;
```

Comment:
Is it intended that if density is `2` then the offset would be one, but if the density was `1.9` then the offset would be `1.9`?

Should these always be integers?

Reply:
Yes, whole numbers only. I updated the comments on density in design system. Was looking for a stronger-typed method of specifying that, but wasn't sure if it was necessary or not.

Reply:
Yea you can strongly type it - we also need to ensure that in code though if that is the rule. While TypeScript is awesome, it is easily possible to circumvent the type system and not everyone that uses these libraries will be using TypeScript. 

You can change this statement to:
`return clamp(Math.floor(designSystem.density)-1, 1)